### ISSUE TITLE
[Bug]: false positive on commonjs imports for no-deprecated 

### ISSUE THREAD

**Issue by jlowcs** [jsx-eslint/eslint-plugin-react#3606]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

I'm getting a `no-deprecated` eslint error for the following:

```js
const { renderToString } = require('react-dom/server');
```

None for ESM imports though:

```js
import { renderToString } from 'react-dom/server';
```

### Expected Behavior

This should not trigger a `no-deprecated` linting error:

```js
const { renderToString } = require('react-dom/server');
```

### eslint-plugin-react version

v7.33.0

### eslint version

v8.45.0

### node version

v18.12.1

**Comment by ljharb** [comment]
Indeed, React.renderToString is what's deprecated, not ReactDOMServer.renderToString.
